common/src/main/java/com/wynntils/core/features/FeatureRegistry.java

```diff
@@ -189,94 +179,5 @@ public Object generate() {
                 return result.toString();
             }
         });
-
-        CrashReportManager.registerCrashContext(new CrashReportManager.ICrashContext() {
-            @Override
-            public String name() {
-                return "Loaded Overlays";
-            }
-
-            @Override
-            public Object generate() {
-                StringBuilder result = new StringBuilder();
-
-                for (Overlay overlay : OVERLAYS) {
-                    if (overlay.isEnabled()) {
-                        result.append("\n\t\t").append(overlay.getTranslatedName());
-                    }
-                }
-
-                return result.toString();
-            }
-        });
-    }
-
-    private static class OverlayListener { // TODO create a enum map for overlays instead of this
-        @SubscribeEvent
-        public static void onTick(ClientTickEvent e) {
-            if (e.getTickPhase() == ClientTickEvent.Phase.END) {
-                for (Overlay overlay : OVERLAYS) {
-                    overlay.tick();
-                }
-            }
-        }
-
-        @SubscribeEvent
-        public static void onRenderPre(RenderEvent.Pre e) {
-            if (!WynnUtils.onServer()) // || !McUtils.mc().playerController.isSpectator())
-            return;
-
-            McUtils.mc().getProfiler().push("preRenOverlay");
-            for (Overlay overlay : OVERLAYS) {
-                if (!overlay.visible) continue;
-                // if (!overlay.active) continue;
-
-                if (overlay.hookElements.length != 0) {
-                    boolean contained = false;
-                    for (RenderEvent.ElementType type : overlay.hookElements) {
-                        if (e.getType() == type) {
-                            contained = true;
-                            break;
-                        }
-                    }
-
-                    if (contained && overlay.visible) {
-                        McUtils.mc().getProfiler().push(overlay.getTranslatedName());
-                        overlay.render(e);
-                        McUtils.mc().getProfiler().pop();
-                    }
-                }
-            }
-
-            McUtils.mc().getProfiler().pop();
-
-            // McIf.mc().getTextureManager().bindTexture(ICONS);
-        }
-
-        @SubscribeEvent
-        public static void onRenderPost(RenderEvent.Post e) {
-            if (!WynnUtils.onServer()) // || !McUtils.mc().playerController.isSpectator())
-            return;
-
-            McUtils.mc().getProfiler().push("postRenOverlay");
```

Comment:
Do you think you can keep the profiler wrappers in the new overlay render code? I think it is nice to have it in place from the beginning, since overlay rendering is one of the ways that Wynntils can have unfortunately large impacts on game performance (rendering is expensive and is done multiple times per second), and it is easy to create bad performance overlays without realizing it.